File: types/sustainability.ts
export type SustainabilityMetricType = 'energy' | 'water' | 'waste' | 'carbon';

export interface SustainabilityMetric {
  id: string;
  zoneId: string;
  metricType: SustainabilityMetricType;
  value: number;
  measuredAt: string;
  createdAt: string;
  updatedAt: string;
}

export interface SustainabilityMetricFormValues {
  zoneId: string;
  metricType: SustainabilityMetricType;
  value: string;
  measuredAt: string;
}

File: backend/schemas/sustainability.py
# backend/schemas/sustainability.py
"""Pydantic schemas for SustainabilityMetric model."""
from __future__ import annotations
import uuid
from datetime import datetime
from pydantic import BaseModel
from backend.models.sustainability_metric import MetricType

class SustainabilityMetricBase(BaseModel):
    zone_id: uuid.UUID
    metric_type: MetricType
    value: float
    measured_at: datetime

class SustainabilityMetricCreate(SustainabilityMetricBase):
    pass

class SustainabilityMetricUpdate(BaseModel):
    zone_id: uuid.UUID | None = None
    metric_type: MetricType | None = None
    value: float | None = None
    measured_at: datetime | None = None


class SustainabilityMetricRead(SustainabilityMetricBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True
